contents/blog/open-source-telemetry-ethical.md

```diff
@@ -0,0 +1,102 @@
+
+Open source projects are zero expectation - they may contain bugs, security risks, or have confusingly outdated documentation. Remember that time you lost hours on a project that just didn't work in the first place?
+
+The fact it's acceptable for open source software not to work is a _great_ thing - it makes it easier for anyone to contribute. Mistakes, or half-baked projects included. You can't have innovation without a lot of people being able to contribute and make mistakes, or give up.
+
+From a utilitarian perspective, we believe that tracking the minimum data possible to build useful technology, in a way that doesn't share that data with 3rd parties, means that you have the right approach to product analytics in your open source project.
+
+From a duty-based perspective, we'd argue that transparency is key to covering intent. Since most people won't assume they're being tracked, that means being upfront in the docs and not relying on people digging through your code.
+
+## Sorting the good from the questionable
+
+First of all, open source is exactly that. That's a big step up from SAAS - you can immediately inspect the code or run it on a computer with no internet connection if needed. However, the precise function of the telemetry has user implications.
+
+### What is being tracked
+
+The first step is to think about how much, if anything, a project really needs to track.
+
+#### Nothing
+
+Nada, zilch. 
+
+If an open source project exists for fun, for one person only, or to get better at a particular technology, then we don't think putting extra tracking in makes sense.
+
+In practice, however, it's pretty much impossible not to track more than you need in this case.
+
+If a project is hosted on GitHub, it automatically provides basic stats (clones / repo visits) provided for all repos.
+
+#### Aggregated tracking
+
+This helps projects validate if the concept of what they're working on is valuable - do people install it? Do they actually use it at all?
+
+It also means they can get a sense of how accessible their software is by looking at how many people work out how to use it. At PostHog, early on, we tracked if people were able to get through the setup process. Many didn't, so we've kept tweaking it - and we still have work to do here :)
+
+With aggregated tracking, a project cannot by definition identify unique users. This means repeat visitors are likely to skew the statistics, and therefore, the understanding of user behavior.
+
+#### Anonymized user tracking
+
+This gives a much stronger signal than the above. It means you'll probably need to start using cookies or some way of obfuscating the user id from their behavior.
+
+It allows projects to see if people who started using the product are coming back, and which parts of the product they're returning too.
+
+#### Full user tracking
+
+Asking permission and not doing this by default feels like the only acceptable way from a user perspective.
```

Comment:
How does that play out with us given that we don't ask for permission? (on the website that is - time to add a banner?)

Reply:
I'd argue the distinction we're making here is this applies for MaryBloggs@example.com being tracked, which doesn't apply to anonymised website visitors. In our case, if we are tracking MaryBloggs@example.com, we do ask for this upfront.

In terms of the execution of GDPR for anonymized users - cookie banners everywhere, I think these have become incredibly annoying to users who *don't* care about this (which I'd argue outweighs the benefit to those who do), and ad blockers are a more effective route. That isn't to say the spirit of GDPR is very good. I'd rather we just obeyed headers appropriately as a starting point, although I'm aware this isn't widely accepted. Think it's a separate issue regardless because of the paragraph above.

Reply:
I see - because on our website we don't grab the email so it's anonymized by default. The user might be tracked but we only tie that info with an email if they sign up for PostHog and opt into tracking. Makes sense.

As for GDPR - I agree with most of the points.  Cookie banners are annoying and GDPR has had bad unintended consequences. However, it's not like websites want to add cookie banners - they _have_ to. I wonder if we fall under that category, but I suspect we might not.